File: Frontend/src/app/viewModels/responses/GeneratedRouteVM.ts
import { RoutePlaceType } from "../Types/RoutePlaceType";

/**
 * View Model that represents a response with a generated route.
 */
export class GeneratedRouteVM {

    /**
     * Construct's an instance of {@link GeneratedRouteVM}
     * 
     * @param name route's name
     * @param startTime route's start date
     * @param finishTime route's finish date
     * @param route route's places and visiting times
     * @param possiblePlaces route's possible palce's replacements 
     */
    constructor(
        public name: string,
        public startTime: Date,
        public finishTime: Date,
        public route: Array<RoutePlaceType>,
        public possiblePlaces: Array<string>
    ) { }
}

File: lei_lds2122_grupo2-main/Backend/Data/ViewModels/Response/GeneratedRouteVM.cs
using Backend.Data.ViewModels.Types;

namespace Backend.Data.ViewModels.Response
{
    /// <summary>
    /// View Model class for the generated route 
    /// </summary>
    public class GeneratedRouteVM
    {
        public string Name { get; set; }
        public DateTime startTime;
        public DateTime finishTime;
        public List<RoutePlaceType> Route { get; set; }
        public List<string> PossiblePlaces { get; set; }

        /// <summary>
        /// Class' constructor
        /// </summary>
        /// <param name="name">Route's name</param>
        /// <param name="route">List of places</param>
        /// <param name="possiblePlaces">List of possible places for the route in case the user wants to regenerate on place</param>
        public GeneratedRouteVM(string name, DateTime startTime, DateTime finishTime, List<RoutePlaceType> route, List<string> possiblePlaces)
        {
            this.Name = name;
            this.startTime = startTime;
            this.finishTime = finishTime;
            this.Route = route;
            this.PossiblePlaces = possiblePlaces;
        }
    }
}